File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>();
        char[] a = s.toCharArray();
        for(char c: a){
            if(c == '('){
                st.push(')');
            }
             else if(c == '['){
                st.push(']');
            }
             else if(c == '{'){
                st.push('}');
            }
            else{
                if(st.size() == 0) return false;
                if(st.pop() != c) return false;
            }
        }
        if(st.size() == 0) return true;
        return false;
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, string: str) -> bool:
        s=[]
        for p in string:
            if p == ')':
                if s and s[-1] == '(':
                    s.pop()
                    continue
                else:
                    return False
            if p == "]":
                if s and s[-1] == '[':
                    s.pop()
                    continue
                else:
                    return False
            if p == "}":
                if s and s[-1] == '{':
                    s.pop()
                    continue
                else:
                    return False
            s.append(p)
        if s:
            return False
        return True
            